### DISCUSSION TITLE
Authorization rule and ${username}

### DISCUSSION THREAD

**Discussion by valdasa** [emqx/emqx#12445] [Q&A] [answered]
So i lets say i have lots of user like "test", "test2", "test3"... etc. and i want to write a rule that allows them to publish and subscribe only to topics that are like there user names

if i write:

{allow, {username, "test"}, all, ["test"]}. 

it works but you wont be doing it for lets say 100 or 1000 users right. So i can get username with ${username} right?

{allow, {username, "${username}"}, all, ["${username}"]}.

But it desn`t work. What am i doing wrong.

**Comment by SergeTupchiy** [comment, marked as answer]
Hi @valdasa,

There is no need to to specify `"${username}"` placeholder in `who` section, placeholders are supported only in topic filters, you can just set `who` = `all`, for example: `{allow, all, all, ["${username}"]}.`

The complete acl file example:
```
{allow, {username, {re, "^dashboard$"}}, subscribe, ["$SYS/#"]}.

{deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.

{allow, all, all, ["${username}"]}.

{deny, all}.
```

**Comment by valdasa** [comment]
thank you, now its obvious :)
